config/config.go

```diff
@@ -483,6 +488,12 @@ func (cfg *Manager) ToJSON() ([]byte, error) {
 	return DefaultJSONMarshal(jcfg)
 }
 
+// EmptyComponents returns a map of component that were found
+// to have no config in JSON file.
+func (cfg *Manager) EmptyComponents() map[string]bool {
```

Comment:
This is not very from elegant, and assumes two components in different sections cannot have the same name...

Rather than this, create a `func IsDefined(SectionType, SectionName) bool`. Carry the cfgMgr into `CreateCluster`